lib/active_merchant/billing/gateways/firstdata_e4_v27.rb

```diff
@@ -304,7 +304,7 @@ def add_address(xml, options)
       def strip_line_breaks(address)
         return unless address.is_a?(Hash)
 
-        Hash[address.map { |k, s| [k, s.tr("\r\n", ' ').strip] }]
+        Hash[address.map { |k, s| [k, s.try(:tr, "\r\n", ' ').strip] }]
```

Comment:
I think the `strip` here will blow up if `s.try...` returns `nil`

Reply:
Good catch! I added a nil value in the strip line test and it failed. Thanks for the 🦅 👀 

Using your recommendation and modifying the test to make sure it doesn't regress.